docs/custom_conf.py

```diff
@@ -334,3 +330,82 @@ def _compute_navigation_tree(context):
     ('py:class', 'ops.testing.CharmType'),
     ('py:obj', 'ops.testing.CharmType'),
 ]
+
+
+# This is very strongly based on
+# https://github.com/sphinx-doc/sphinx/blob/03b9134ee00e98df4f8b5f6d22f345cdafe31870/sphinx/domains/changeset.py#L46
+# Unfortunately, the built-in class is not easily subclassable without also
+# requiring extra CSS.
```

Comment:
This is okay, but would it be easier to add said extra CSS?

Reply:
The code would then be able to be roughly what's in this [earlier commit](https://github.com/canonical/operator/pull/1311/commits/5476a0c00c3b9fc27f7e5b00ebeeb9f2018b23bb).

Then we'd need a custom CSS file in `docs/.sphinx/_static`, which I think could just have something like:

```css
div.jujuversion {
    border-color: #38a84d;
}
div.jujuversion p {
    margin-bottom: .125rem;
    margin-top: .125rem;
}
div.jujuversion .versionmodified {
    color: #21632c;
}
```

And the CSS would be added to the `custom_html_css_files` variable in `custom_conf.py`.

Our current method of pulling in changes from the starter pack involves replacing the `.sphinx` folder entirely, so we'd need to handle that somehow (either instructions in `HACKING.md`, or having tox copy the file there from somewhere safer). I'm hopeful that the moves towards [using an extension](https://discourse.canonical.com/t/this-month-in-documentation-july-2024/4045#starter-pack) will make everything much simpler for us in that regard, so perhaps this is only a short-term issue.

I'm not sure which I prefer (I don't really love either). Thoughts?

Reply:
> Our current method of pulling in changes from the starter pack involves replacing the .sphinx folder entirely

Okay, given that, let's stick with what you have.